content/en/users/data/management/datahub/api/_index.md

```diff
@@ -218,60 +222,82 @@ The following variables should be set:
-  "$ONEPROVIDER_HOST/api/v3/oneprovider/shares-id/$DIR_ID_TO_SHARE" | jq '.'
+$ DIR_ID_TO_SHARE=<DIR_ID>
+$ curl -sS --tlsv1.2 -H "X-Auth-Token: $API_ACCESS_TOKEN" \
+    -X POST -H 'Content-Type: application/json' \
+    -d '{"name": "input"}'
+    "$ONEPROVIDER_HOST/api/v3/oneprovider/shares-id/$DIR_ID_TO_SHARE" | jq '.'
 
 # Displaying the share
-SHARE_ID=<SHARED_ID>
-curl -sS --tlsv1.2 -H "X-Auth-Token: $API_ACCESS_TOKEN" \
-   "$ONEZONE_HOST/api/v3/onezone/shares/$SHARE_ID" | jq '.'
+$ SHARE_ID=<SHARED_ID>
+$ curl -sS --tlsv1.2 -H "X-Auth-Token: $API_ACCESS_TOKEN" \
+    "$ONEZONE_HOST/api/v3/onezone/shares/$SHARE_ID" | jq '.'
 
 # Registering a handle
 # Proper Dublin Core metadata is required
 # It can be created using https://nsteffel.github.io/dublin_core_generator/generator_nq.html
-cat metadata.xml
+$ cat metadata.xml
 # Escape double quotes and drop line return
-METADATA=$(cat metadata.xml | sed 's/"/\\"/g' | tr '\n' ' ')
+$ METADATA=$(cat metadata.xml | sed 's/"/\\"/g' | tr '\n' ' ')
 # On handle creation the created handles is provided in the Location header
-curl -D - --tlsv1.2 -H "X-Auth-Token: $API_ACCESS_TOKEN" \
-  -H "Content-type: application/json" -X POST \
-  -d '{"handleServiceId": "'"$HANDLE_SERVICE_ID"'", "resourceType": "Share", "resourceId": "'"$SHARE_ID"'", "metadata": "'"$METADATA"'"}' \
-  "$ONEZONE_HOST/api/v3/onezone/user/handles"
+$ curl -D - --tlsv1.2 -H "X-Auth-Token: $API_ACCESS_TOKEN" \
+    -H "Content-type: application/json" -X POST \
+    -d '{"handleServiceId": "'"$HANDLE_SERVICE_ID"'", "resourceType": "Share", "resourceId": "'"$SHARE_ID"'", "metadata": "'"$METADATA"'"}' \
+    "$ONEZONE_HOST/api/v3/onezone/user/handles"
 
 # Listing handles
-curl --tlsv1.2 -H "X-Auth-Token: $API_ACCESS_TOKEN" \
-  "$ONEZONE_HOST/api/v3/onezone/user/handles"
+$ curl --tlsv1.2 -H "X-Auth-Token: $API_ACCESS_TOKEN" \
+    "$ONEZONE_HOST/api/v3/onezone/user/handles"
 
 # Displaying a handle
-HANDLE_ID=<HANDLE_ID>
-curl --tlsv1.2 -H "X-Auth-Token: $API_ACCESS_TOKEN" \
-  "$ONEZONE_HOST/api/v3/onezone/user/handles/$HANDLE_ID"
+$ HANDLE_ID=<HANDLE_ID>
+$ curl --tlsv1.2 -H "X-Auth-Token: $API_ACCESS_TOKEN" \
+    "$ONEZONE_HOST/api/v3/onezone/user/handles/$HANDLE_ID"
 ```
-<!-- markdownlint-enable line-length -->
+
+## Subscribe to file events
+
+Following is an example of how to subscribe to DataHub to receive notification
+on file events which is described in details in the official documentation
+[Subscribe to file events](https://onedata.org/#/home/api/stable/oneprovider?anchor=operation/get_space_changes):
+
+```shell
+$ curl -N -H "X-Auth-Token: $TOKEN" \
+    -X POST "https://$HOST/api/v3/oneprovider/changes/metadata/$SPACE_ID" \
+    -H "Content-Type: application/json" -d "@./changes_req.json"
+```
+
+This requires the permission set as following:
+
+![Viewing file popularity for smart caching](datahub-space-permissions.png)
+
+For groups or single users. For single users, one way to add one is to select
```

Comment:
what we are doing so far is a kind of workaround..i need to check with onedata guys what is the best way to achieve this

Reply:
Shall  we keep it anyway for now as it has already been used twice and at the moment is not really documented anywhere. If a better way comes out maybe we can then update this.

Reply:
ok let's do as you suggest, and update it with new instructions when available